src/do-not-disturb.js

```diff
@@ -0,0 +1,53 @@
+
+// Always false if the platform doesn't support it.
+global.isDoNotDisturb = false;
+
+const LINUX_SETTING_COMMAND = "gsettings get org.gnome.desktop.notifications show-banners";
+
+function init() {
+    if (process.platform === "linux") {
+        return initForLinux();
+    }
+}
+
+async function initForLinux() {
+    // TODO: This is specific to the GNOME desktop implementation of DND
+
+    const DBus = require('dbus-next');
+    const { execSync } = require('child_process');
+    // First we need to determine the value of dnd.
+    try {
+        // XXX: After much flailing about, I couldn't find another acceptable way to fetch this setting value.
+        const value = execSync(LINUX_SETTING_COMMAND, { encoding: "utf-8" }).trim();
```

Comment:
We have async/await now so there shouldn't really be any reason to use the sync functions